docs/core/deploying/native-aot/fixing-warnings.md

```diff
@@ -0,0 +1,80 @@
+
+To prevent changes in behavior when deploying as native AOT, the .NET SDK provides static analysis of AOT compatibility through "AOT warnings." AOT warnings are produced when the build finds code that may not be compatible with AOT. Code that's not AOT-compatible may produce behavioral changes, or even crashes, in an application after it has been built as native AOT. Ideally, all applications that use native AOT should have no AOT warnings. If there are any AOT warnings, the app should be thoroughly tested after building as native AOT to ensure that there are no behavior changes.
+
+## Examples of AOT warnings
+
+For most C# code, it's straightforward to determine what native code needs to be generated. The native compiler can walk the method bodies and find what native code and data structures are accessed. Unfortunately, some features, like reflection, present a significant problem. Consider the following code:
+
+```csharp
+Type t = typeof(int);
+while (true)
+{
+    t = typeof(GenericType<>).MakeGenericType(t);
+    Console.WriteLine(Activator.CreateInstance(t));
+}
+
+struct GenericType<T> { }
+```
+
+While the above program is not very useful, it represents an extreme case that requires an infinite number of generic types to be created when building the application as native AOT. Without native AOT, the program would run until it runs out of memory. With native AOT, we would not be able to even build it if we were to generate all the necessary types (the infinite number of them).
+
+In this case, native AOT build will issue following warning on the `MakeGenericType` line:
+
+```
+AOT analysis warning IL3050: Program.<Main>$(String[]): Using member 'System.Type.MakeGenericType(Type[])' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. The native code for this instantiation might not be available at runtime.
+```
+
+At run time, the application will indeed throw an exception from the `MakeGenericType` call.
+
+## Reacting to AOT warnings
+
+The AOT warnings are meant to bring predictability to native AOT builds. Majority of AOT warning are about possible run time exception in situations when native code wasn't generated to support the scenario. The broadest category is `RequiresDynamicCodeAttribute`.
+
+### RequiresDynamicCode
+
+<xref:System.Diagnostics.CodeAnalysis.RequiresDynamicCodeAttribute> is simple and broad: it's an attribute that means the member has been annotated incompatible with AOT, meaning that it might use reflection or some other mechanism to create new native code at runtime. This attribute is used when code is fundamentally not AOT compatible, or the native dependency is too complex to statically predict at build time. This would often be true for methods that use the Type.MakeGenericType API, reflection emit, or other runtime code generation technologies. An example would be:
```

Comment:
```suggestion
<xref:System.Diagnostics.CodeAnalysis.RequiresDynamicCodeAttribute> is simple and broad: it's an attribute that means the member has been annotated as being incompatible with AOT. This annotation means that the member might use reflection or another mechanism to create new native code at run time. This attribute is used when code is fundamentally not AOT compatible, or the native dependency is too complex to statically predict at build time. This would often be true for methods that use the `Type.MakeGenericType` API, reflection emit, or other run-time code generation technologies. The following code shows an example.
```